examples/chat.rs

```diff
@@ -234,10 +239,21 @@ impl SignedMessage {
     }
 }
 
+/// A chat message.
 #[derive(Debug, Serialize, Deserialize)]
 enum Message {
-    AboutMe { name: String },
-    Message { text: String },
+    AboutMe {
+        name: String,
+    },
+    /// A chat message.
+    ///
+    /// `nonce` is used to uniquely identify each message - iroh-gossip deduplicates
+    /// based on the hash of the message, so we need to ensure that the message is
+    /// unique, so two identical chat messages are still delivered.
+    Message {
+        text: String,
+        nonce: u32,
```

Comment:
This is good! Can you make it a `[u8; 16]` though, just to make it borderline practically impossible for colliding nonces for the price of just a couple more bytes to send?

```suggestion
        nonce: [u8; 16],
```